### ISSUE TITLE
Can I limit the number of lines for every point?

### ISSUE THREAD

**Issue by KitDevUA** [VincentGarreau/particles.js#247]
I want to make the effect of floating constellations. For example, every point has a maximum of 2 lines with the nearest points.

**Comment by mclark-newvistas** [comment]
Yes - see https://github.com/VincentGarreau/particles.js/blob/master/particles.js#L676. Change it to add/increment a value on both particles and drop out early if either value exceeds a limit.

However - it doesn't check for the _closest_ particle (and doing so would be slower). Possibly the best option with the code as is would be to restructure to create a list of lines w/lengths, sort that and chop above a limit, _then_ draw the lines. That would be pretty simple.

Best might be to implement a [k-d tree](https://en.wikipedia.org/wiki/K-d_tree) for searching points - but that'd be pretty bothersome :).
